### PR TITLE
#6658: Add code editor to JS Brick

### LINKED ISSUE #6658: Slice 1.2: Use the Ace Editor as the widget to provide line numbers and syntax highlighting for  Javascript
https://www.notion.so/pixiebrix/Add-Javascript-brick-for-data-transformation-to-avoid-JQ-754618ab6bc64a868b149f2a8c389454?pvs=4#e110ef9dd5e8487980410db4b957c595

### PR DESCRIPTION
**Pull request by grahamlangford**
## What does this PR do?

- Closes #6658 

## Demo

https://www.loom.com/share/e74c8fda857a40d09bed9aa5cf248b4b

## Checklist

- [x] Add tests
- [x] New files added to `src/tsconfig.strictNullChecks.json` (if possible)
- [x] Designate a primary reviewer @twschiller

### CODE REVIEW

File: src/components/fields/schemaFields/widgets/TextWidget.stories.tsx
Comment (grahamlangford) [inline src/components/fields/schemaFields/widgets/TextWidget.stories.tsx:1]: I used this as a template for creating the CodeEditorWidget story, but first I updated it to no longer use deprecated types.

File: src/bricks/transformers/javascript.ts [hunk 1]
```diff
@@ -60,6 +61,12 @@ export class JavaScriptTransformer extends TransformerABC {
     ["function"]
   );
 
+  uiSchema: UiSchema = {
+    function: {
+      "ui:widget": "CodeEditorWidget",
```
Comment (twschiller) [inline src/bricks/transformers/javascript.ts:66]: ❤️

File: src/vendors/AceEditorSync.tsx
Comment (grahamlangford) [inline src/vendors/AceEditorSync.tsx:1]: TODO: determine if we need to split these imports out for bundle size/performance reasons

Comment (grahamlangford) [reply src/vendors/AceEditorSync.tsx:1]: image
Based on the visualizer, looks like the additional deps add only a fraction of the overall size of the bundle. Doesn't seem worth it to split out at this time.

**Review by twschiller — APPROVED** [review]

**Pull request merged**